### ISSUE TITLE
Convert small data sets to "static" versions

### ISSUE THREAD

**Issue by cbeleites** [cbeleites/hyperSpec#137]
Currently, **hyperSpec** generates its example data sets from real spectra files. 
This is done during the building of the vignettes and comes at the cost of a non-standard building process with complex `Makefile`s (#132).

As part of the general clean up 

- [x] make "static" snapshots of the data sets the *primary* source, i.e. place them under version control. 
  - [x] `barbiturates`
  - [x] `flu`
  - [x] `laser`
  - [x] `paracetamol`
- [x] if possible, prune the data to be smaller (`barbiturates`, `laser`?)
- [x] clarify that the example data may not be exactly the same as the data in the corresponding vignettes where the original spectra files are imported.
- [x] clarify what kind of data they exemplify
- [x] put code to generate the data and possibly the raw/binary source sets into `data-raw/`. List `data-raw/` in `.Rbuildignore`.  
see: https://r-pkgs.org/data.html#data-data
- [x] delete the parts of the `Makefile`s that create and copy these files into the package directory tree
- [x] delete the `.gitignore` entries refering to them.

**Comment by bryanhanson** [comment]
On the third point, why not use the static data in the vignettes?  That would simplify things a great deal.

**Comment by cbeleites** [comment]
@bryanhanson: thanks for spotting. I was thinking of the vignettes that import the real spectra. 

(Future) vignettes that are shipped with **hyperSpec** can and should use the data sets as in hyperSpec. 

related: #138

**Comment by GegznaV** [comment]
Why is `chondro` not on this list?

The description of the datasets may contain sentence, such as "The dataset is provided with the package is a subset of... (some real data)" To illustrate the ideas/functionality, we can even use artificial spectra. Just a thought, but maybe, some functions that simulate spectra could be included?

**Comment by GegznaV** [comment]
> (Future) vignettes that are shipped with **hyperSpec** can and should use the data sets as in hyperSpec.

I agree. But he links to the original dataset could also be provided.

**Comment by cbeleites** [comment]
`chondro` is not on this list, because it needs to be dealt with in a different manner since it is too big (#129 ). => the solution for `chondro` is that 

- For the purpose of examples in the help pages, @bryanhanson's new `FauxCell` will take the role of `chondro` to illustrate a spectral map/image. 
- @ximeg is preparing a separate data package that contains the original data of `chondro`. The vignette (naturally) moves into the data package. The vignette already now depends on the original data: the PCA compressed version shows substantial artifacts (of the PCA compression) that mess up the workflow. But the best lossless compression of the original data we can easily use (xz) is still several MB of data. So no way to ship this other than in a data package.

**Comment by cbeleites** [comment]
> links to the original dataset could also be provided.

There is a `@source` Roxygen parameter that can be used. I was thinking of putting something along the lines e.g. for `barbiturates` "This data set was prepared from the first few subfiles of "BARBITURATES.SPC"` - for more information and the original file, see package 'hyperSpec_import_spc'"

**Comment by GegznaV** [comment]
> see package 'hyperSpec_import_spc'"

Are the names of the new packages confirmed? Was there a discussion on that? I would prefer to have some shorter names as `hy.import`,  `hy.manager`, `hy.plot`, `hyperImport`, `hyperWrangler` `hyperPlot`, etc.

**Comment by cbeleites** [comment]
no, that discussion was postponed to the next video call (Monday 7 pm EEST).

I opened an issue, though: #140

**Comment by GegznaV** [comment]
I'm doing some experimentation on this topic. There are some _(in my opinion)_ unnecessary entries in `.Rbuildignore` or/and in `.gitignore`, which may have led to unsuccessful results of Claudia's experimentation with data (she talked on that this Monday).

**Comment by cbeleites** [comment]
I had not been experimenting with the data sets in this issue but with `fauxCell`. The small data sets here are unproblematic. They are standard `.rda` files already, and behave as expected. `fauxCell` is totally different since it is a variable created by the package source code. Explanation on what I tried is at the end of #114 .

You are right that `.gitignore` will need to be changed: at the moment it reflects which files are created or copied into their place in the package directory tree by `make`. 
I added cleaning up `.gitignore` and the `Makefile`s to the TODO list at the top of the issue.
